### PR TITLE
Add Positive number function to the main

### PR DESCRIPTION
**Pull request by Luyando-Chitindi**
---
name: positive number function solution review 
about: A template PR for code review with a checklist
---

function that checks if number is positive

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/is_positive.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on XX XX XX
```
Comment (Muna-S) [inline solutions/is_positive.py:4]: So I have checked the is_positive.py, everything seems fine - though just in this line, maybe you forgot to replace XX... with the date?

File: solutions/tests/test_is_positive.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+"""
+
+import unittest
+from solutions.is_positive import is_positive
+
+
+class TestIsPositive(unittest.TestCase):
+    """Test the is_positive function"""
+
+    def test_positive_number(self):
```
Comment (Muna-S) [inline solutions/tests/test_is_positive.py:16]: (Just a suggestion for conciseness, but nothing wrong here) we can be a bit more concise if we write: 
def test_positive_number(self):
   self.assertTrue(is_positive(4))
and maybe you can apply this to the two after as well. but I am not sure if my suggestion misses something from your original code.

File: solutions/is_positive.py (hunk 1 above)
Comment (Luyando-Chitindi) [reply solutions/is_positive.py:4]: Hi Muna,
Thanks so much for this comment. I really appreciate! 

Um, in as much as this is just a basic function, in the future, I'd like to build upon it, so that's why I haven't put a specific date yet. Since this project is primarily a test to implement basic functions, I decided to use something simple. I envision using this function as a building block for more complex operations in my own projects. For instance, this positive number check could be integrated into a larger program for data analysis (which is a topic that interests me).

File: solutions/tests/test_is_positive.py (hunk 2 above)
Comment (Luyando-Chitindi) [reply solutions/tests/test_is_positive.py:16]: Hi Muna,

Thanks for the suggestion! I understand your point about conciseness, and it's actually much faster! Thank you so much.
I intentionally wrote the test cases with separate lines for actual and expected values to make the test logic clearer and easier to read.

To be honest, as I'm still getting familiar with testing and coding best practices, this format really helps me understand my own thought process and debug more efficiently. It's a bit like training wheels for my coding skills.

But you are very much right! 
I appreciate your feedback and will definitely use your concise approach in my future codes as I become more comfortable with testing.

Thank you Muna :)

**Comment by Muna-S** [comment]
**Important note on why comments where resolved without change**: Luyando will be unable to access a laptop, so I marked my comments as resolved - HOWEVER, her code was perfect and my 2 comments were merely suggestions, so it does not really impact her project. In other words, whether or not changed were applied based on my comments, the code would work as perfectly as ever.

**Review by Muna-S — APPROVED** [review]
Code performs its function well, and passes all checks.
